README_Building.md

```diff
@@ -35,6 +35,241 @@ It may complain that there isn't enough space on the chip. This isn't an issue u
+* Your board will now be flashed
+
+**Note:** if you want to you can change permissions so you don't need `sudo` by typing `sudo cp misc/45-espruino.rules /etc/udev/rules.d;sudo udevadm control --reload-rules` and then re-inserting the board.
+
+Building
+========
+
+**Note:** 
+- If you're swapping between compiling for different targets, **you
+need to call `make clean`** before you compile for the new target.
+- If you have a ld error, double check the board name in the <BOARD>=1 make
+
+Under Linux
+-----------
+
+Espruino is easy to build under Linux, and it is possible to build under MacOS with some effort. If you don't have Linux it's **much** easier to install it in a Virtual Machine. See the heading **Building under Windows/MacOS with a VM** below for more information.
+
+### for STM32 Boards (incl. [Espruino Board](http://www.espruino.com/EspruinoBoard))
+
+The (previously suggested) CodeSourcery GCC compiler is no longer available. We'd suggest you use [gcc-arm-none-eabi](https://launchpad.net/gcc-arm-embedded/+download).
+
+Download the compiler, set up your path so you have access to it, and run:
+
+```YOUR_BOARD_NAME=1 RELEASE=1 make```
+
+* See the top of Makefile for board names
+* Without `RELEASE=1`, assertions are kept in the code (which is good for debugging, bad for performance + code size)
+* `BOARDNAME=1 RELEASE=1 make serialflash` will flash to /dev/ttyUSB0 using the STM32 serial bootloader (what's needed for the Espruino and HY boards)
+* `BOARDNAME=1 RELEASE=1 make flash` will flash using st-flash if it's a discovery board, the maple bootloader if using that board, or will copy the binary to `/media/NUCLEO` if using a Nucleo board.
+
+It may complain that there isn't enough space on the chip. This isn't an issue unless you save to flash, but you can fix the error in a few ways:
+
+* Disable the check by adding `TRAVIS=1`
+* Change the compile flags from `-O3` to `-Os` in the `Makefile`
+* Knock out some functionality (like `USE_GRAPHICS=1`) that you don't need in the `Makefile`
+* Try different compilers. `codesourcery-2013.05-23-arm-none-eabi` provides low binary size for `-O3`
+
+**Note:** Espruino boards contain a special bootloader at `0x08000000` (the default address), with the Espruino binary moved on 10240 bytes to `0x08002800`. To load the Espruino binary onto a board at the correct address, use `ESPRUINO_1V3=1 RELEASE=1 make serialflash`. If you want to make a binary that contains the bootloader as well as Espruino (like the ones that the Espruino Web IDE expects to use) use the script `scripts/create_espruino_image_1v3.sh` which will compile the bootloader *and* Espruino, and then join them together.
+
+----
+
 ### for Nordic Semiconductor's nRF51/nRF52 series devices
 
 The (previously suggested) CodeSourcery GCC compiler is no longer available. We'd suggest you use [gcc-arm-none-eabi](https://launchpad.net/gcc-arm-embedded/+download).
```

Comment:
Could be I have made a mistake in PR> I will relate your comment to what I have done and get back on this